src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -189,11 +189,36 @@ private function flatten(array $array, array &$result, $keySeparator, $parentKey
             if (\is_array($value)) {
                 $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
             } else {
-                $result[$parentKey.$key] = $value;
+                $result[$parentKey.$key] = $this->cleanValueByType($value);
             }
         }
     }
 
+    /**
+     * Ensures an actual value is used instead of a blank value when dealing
+     * with true, false and null (like a nullable bool in a database).
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    private function cleanValueByType($value)
+    {
+        $clean = $value;
+
+        if (false === $value) {
```

Comment:
I would use early return no ?
```
if (false === $value) {
   return 0;
}

if (null === $value) {
   return ' ';
}

if (true === $value) {
   return 1;
}

return $value;
```

Reply:
Thanks @kevin-verschaeve, I've applied that change now